Orange/widgets/unsupervised/owkmeans.py

```diff
@@ -584,8 +599,18 @@ def set_data(self, data):
         self.controls.normalize.setDisabled(
             bool(self.data) and sp.issparse(self.data.X))
 
+        if type(self.data) is not type(old_data):
+            if isinstance(self.data, DaskTable):
+                self.INIT_METHODS = self.EXTRA_INIT_METHODS + self.BASE_INIT_METHODS
+                self.init_method.insertItem(0, self.EXTRA_INIT_METHODS[0][0])
+                self.init_method.setCurrentIndex(self.smart_init)
+            elif old_data is not None:
+                self.INIT_METHODS = self.BASE_INIT_METHODS
+                self.init_method.removeItem(0)
+                self.smart_init = self.init_method.currentIndex()
+
         # Do not needlessly recluster the data if X hasn't changed
-        if old_data and self.data and array_equal(self.data.X, old_data.X):
+        if old_data and self.data and array_equal(self.data.X, old_data.X):  # could be an issue for dask
```

Comment:
This is a good find. I remember you telling me about it, but then I forgot. I do not have a good solution for now, but we would need to ensure (and test) at the minimum that `array_equal` does not load the whole data set into memory.

Reply:
Turns out np.allclose works fine on dask. Comparing an 8GB array against itself took a little over two seconds (this is worst case scenario, runing in an ipython shell that doesn't play well with dask) and memory usage stayed very much within reasonable limits.